src/storage/v2/storage.hpp

```diff
@@ -309,11 +310,11 @@ class Storage final {
 
     void AdvanceCommand();
 
-    /// Commit returns `ConstraintViolation` if the changes made by this
-    /// transaction violate an existence or unique constraint. In that case the
-    /// transaction is automatically aborted. Otherwise, void is returned.
+    /// Commit returns `StorageError` if the changes made by this transaction violate an existence, unique constraint or
+    /// data could NOT be replicated to SYNC replica. In that case of existence/unique constraint the transaction is
+    /// automatically aborted. Otherwise, void is returned.
     /// @throw std::bad_alloc
-    utils::BasicResult<ConstraintViolation, void> Commit(std::optional<uint64_t> desired_commit_timestamp = {});
+    utils::BasicResult<StorageError, void> Commit(std::optional<uint64_t> desired_commit_timestamp = {});
```

Comment:
Reminder to add no discard keyword to all new methods as well to ensure non calls have been missed